spec/lib/services/lookup-spec.js

```diff
@@ -100,17 +101,16 @@ describe('Lookup Service', function () {
 
         it('should allow multple simultaneous cache checks', function () {
             expect(lookupService.checkNodeIdCache('testAddress')).to.be.null;
-            lookupService.checkNodeIdCache('testAddress').then(spy1, spy1);
-            lookupService.checkNodeIdCache('testAddress').then(spy2, spy2);
+            promise1 = lookupService.checkNodeIdCache('testAddress');
+            promise2 = lookupService.checkNodeIdCache('testAddress');
         });
 
-        it('should resolve pending cache checks once a value is assigned', function (done) {
+        it('should resolve pending cache checks once a value is assigned', function () {
             lookupService.assignNodeIdCache('testAddress', 'nodeId');
-            setTimeout(function () {
-                expect(spy1.called).to.be.ok;
-                expect(spy2.called).to.be.ok;
-                done();
-            }, 0);
+            return Promise.all([
```

Comment:
Actually, the whole unit-test was written in the dependency way you mentioned, and even worse, there is a large number of other unit-tests are also written in the same way. To break the dependency between each test case is not the intention of my PR, such refactor also need a lot time and code change.

Luckily the mocha runs the test cases in sequence, so there is no problem for such kind of design.